src/provider/provider_coarse.c

```diff
@@ -347,6 +611,11 @@ static enum umf_result_t coarse_memory_provider_initialize(void *params,
         coarse_params->upstream_memory_provider;
     coarse_provider->block_list = NULL;
     coarse_provider->alloc_list = NULL;
+    coarse_provider->free_blocks =
+        ravl_new_sized(ravl_comp, sizeof(ravl_data_t));
```

Comment:
please add checks for free_blocks and allocated_blocks != NULL and handle error

Reply:
Done